app.js

```diff
@@ -13,6 +13,49 @@ function hasFilter(name, filters) {
   return false;
 }
 
+function stateToUrl(state) {
+  var qsParams = Object.keys(state);
+
+  if (!qsParams.length) {
+    return "";
+  }
+
+  return "?" + qsParams.map(function(param) {
+    return param + "=" + state[param];
+  }).join("&");
+}
+
+function initialSetStateByUrl() {
+  if (!location.search) {
+    return setDefaultState();
+  }
+
+  var state = location.search.replace("?", "").split('&')
+    .map(function(search) { return search.split("="); })
+    .reduce(function(acc, item) {
+      acc[item[0]] = item[1];
+      return acc;
+    }, {});
+
+  history.replaceState(state, "", stateToUrl(state));
+}
+
+function setDefaultState() {
+  setQueryString("good-first", true);
```

Comment:
It would be nice if we could pass an object to `setQueryString`, like 
```js
setQueryString({
  "good-first": true,
  "tool": "all"
});
```
It feels a bit weird to have to call the function twice.